x/emissions/module/rewards/topic_rewards.go

```diff
@@ -104,13 +106,57 @@ func SafeApplyFuncOnAllRewardReadyTopics(
 	return nil
 }
 
-// Iterates through every reward-ready topic, computes its target weight, then exponential moving average to get weight.
+// "Churn-ready topic" is active, has an epoch that ended, and is in top N by weights, has non-zero weight
+func ApplyFuncOnAllChurnReadyTopics(
+	ctx context.Context,
+	k keeper.Keeper,
+	block BlockHeight,
+	fn func(ctx context.Context, topic *types.Topic) error,
+	topicPageLimit uint64,
+	maxTopicPages uint64,
+) error {
+	weights, _, _, err := GetTopicWeights(ctx, k, block, false, false)
+	if err != nil {
+		return errors.Wrapf(err, "weights error")
+	}
+	// Sort remaining active topics by weight desc and skim the top via SortTopicsByReturnDescWithRandomTiebreaker() and param MaxTopicsPerBlock
+	maxTopicsPerBlock, err := k.GetParamsMaxTopicsPerBlock(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get max topics per block")
+	}
+	weightsOfTopActiveTopics := SkimTopTopicsByWeightDesc(weights, maxTopicsPerBlock, block)
+
+	for topicId, weight := range weightsOfTopActiveTopics {
+		if weight.Equal(alloraMath.MustNewDecFromString("0")) {
+			fmt.Println("Skipping Topic ID: ", topicId, " Weight: ", weight)
+			continue
+		}
+		// Get the topic
+		topic, err := k.GetTopic(ctx, topicId)
+		if err != nil {
+			fmt.Println("Error getting topic: ", err)
+			continue
+		}
+		// Execute the function
+		err = fn(ctx, &topic)
```

Comment:
I'd name this function instead

Reply:
Not sure what you mean... "fn" can be any function to execute, so it has to have a general name.

Reply:
diego is right